octavia_f5/controller/worker/l2_sync_manager.py

```diff
@@ -202,13 +202,13 @@ def ensure_l2_flow(self, selfips: [network_models.Port], network_id: str, device
 
             selfips_for_host = [selfip for selfip in selfips if bigip.hostname in selfip.name]
             subnet_ids = set(sip.fixed_ips[0].subnet_id for sip in selfips_for_host)
-            ensure_l2_flow_data[bigip.hostname] = {
+            ensure_l2_flow_data[bigip] = {
                 'store': {'bigip': bigip, 'network': network, 'subnet_id': subnet_ids.pop()},
                 'selfips': selfips_for_host,
             }
         fs[self.executor.submit(
             self._do_ensure_l2_flow,
-            data=ensure_l2_flow_data)] = ','.join(ensure_l2_flow_data.keys())
+            data=ensure_l2_flow_data)] = ensure_l2_flow_data.keys()
```

Comment:
I would rather use the additional list to save filtered bigips objects than putting objects into the dict keys, which is usually a bad idea because `__hash__` and `__eq__` functions are not implemented in these objects correctly, but in this specific case when we are not using keys anywhere else - it's OK.

Reply:
Yeah, I also find it a bit icky, but I consciously decided against refactoring that issue in this PR to get this fix through first. It's definitely a good issue for refactoring in the future.
If you can come up with a quick way to refactor it with a very small diff, feel free to make another stacked PR.

Reply:
Done: https://github.com/sapcc/octavia-f5-provider-driver/pull/292

Reply:
This solution is working for me as well. Approved